astoria/consumers/astwifid.py

```diff
@@ -0,0 +1,164 @@
+"""
+Astoria WiFi Daemon.
+
+Manages a WiFi hotspot for the robot.
+"""
+import asyncio
+import logging
+import subprocess
+from subprocess import check_output, Popen
+from json import loads, JSONDecodeError
+from pathlib import Path
+from typing import Match, Optional
+
+import click
+from pydantic import ValidationError
+
+from astoria.common.consumer import StateConsumer
+from astoria.common.messages.astmetad import Metadata, MetadataManagerMessage
+
+LOGGER = logging.getLogger(__name__)
+
+loop = asyncio.get_event_loop()
+
+
+@click.command("astwifid")
+@click.option("-v", "--verbose", is_flag=True)
+@click.option("-c", "--config-file", type=click.Path(exists=True))
+def main(*, verbose: bool, config_file: Optional[str]) -> None:
+    """WiFi Daemon Application Entrypoint."""
+    wifid = WiFiHotspotDaemon(verbose, config_file)
+    loop.run_until_complete(wifid.run())
+
+
+class WiFiHotspotDaemon(StateConsumer):
+    name = "astwifid"
+
+    dependencies = ["astmetad"]
+
+    def _init(self) -> None:
+        self._lifecycle: Optional[WiFiHotspotLifeCycle] = None
+
+        self._mqtt.subscribe("astmetad", self.handle_astmetad_message)
+
+    async def main(self) -> None:
+        """Main routine for astwifid."""
+        # Wait whilst the program is running.
+        await self.wait_loop()
+
+        # Stop the hotspot when we shutdown
+        if self._lifecycle:
+            await self._lifecycle.stop_hotspot()
+
+    @property
+    def name_prefix(self) -> str:
+        return "astwifid"
+
+    async def handle_astmetad_message(
+            self,
+            match: Match[str],
+            payload: str,
+    ) -> None:
+        if payload:
+            try:
+                metadata_manager_message = MetadataManagerMessage(**loads(payload))
+                await self.handle_metadata(metadata_manager_message.metadata)
+            except ValidationError:
+                LOGGER.warning("Received bad metadata manager message.")
+            except JSONDecodeError:
+                LOGGER.warning("Received bad JSON in metadata manager message.")
+        else:
+            LOGGER.warning("Received empty metadata manager message.")
+
+    async def handle_metadata(self, metadata: Metadata) -> None:
+
+        if self._lifecycle:
+            if metadata.wifi_enabled:
+                # Check if creds match
+                pass
+            else:
+                # Turn it off!
+                await self._lifecycle.stop_hotspot()
+                self._lifecycle = None
+        else:
+            if metadata.wifi_enabled:
```

Comment:
I'd suggest replacing both of these `if metadata.wifi_enabled` statements with an if to a function call that checks more stuff.

e.g

```python
if metadata.is_wifi_valid():
```

and 

```
class Metadata:
...python
    def is_wifi_valid(self) -> bool:
        return self.wifi_enabled and self.wifi_ssid is not None and self.wifi_psk is not None and self.wifi_region is not None
```